src/java/src/main/java/com/dnanexus/DXHTTPRequest.java

```diff
@@ -362,48 +361,21 @@ private ParsedResponse requestImpl(String resource, String data, boolean parseRe
                     throw DXAPIException.getInstance(errorType, errorMessage, statusCode);
                 } else {
                     // Propagate 500 error to caller
-                    if (this.disableRetry && statusCode != 503) {
+                    if (this.disableRetry) {
                         logError("POST " + resource + ": " + statusCode + " Internal Server Error, try "
                                 + String.valueOf(attempts + 1) + "/" + NUM_RETRIES
                                 + " Request ID: " +  requestId);
                         throw new InternalErrorException("Internal Server Error", statusCode);
                     }
-                    // If retries enabled, 500 InternalError should get retried unconditionally
+                    // If retries enabled, 500 InternalError should get retried unconditionally,
+                    // and so should 503 ServiceUnavailable (TIP-1023).
                     retryRequest = true;
-                    if (statusCode == 503) {
```

Comment:
Here, the response from the server was consume when the status code is 503. We are removing this functionality in this commit, do we want to remove it?

The thing is that if the header didn't contain (or contained malformed) "retry-after" we set retry to be the default 60 seconds. Maybe instead we should set it to exponential backoff here? We'd keep reading the "retry-after" header but at the same time we'd have more clever retry with exponential backoff.

Reply:
Is the retry-after header always included?

Reply:
Also, what is the behavior in the Python bindings? We should synchronize the implementations. The support for disabling should not impact the default behavior right?

Reply:
@ptn24 It's not always included and we handle that [here](https://github.com/dnanexus/dx-toolkit/commit/968fdd67433dcea7c6b7eb14ce2536991829b9ab#diff-6122e848fa81a7baae8083db2bb6c3bdR338)